operations/security_and_authentication/Add_LDAP_User_Federation.md

```diff
@@ -1,263 +1,371 @@
+        - default: shasta
+        - type: string
+   ```
+
+4. Prepare to generate Sealed Secrets.
+   
+   Secrets are stored in customizations.yaml as `SealedSecret` resources 
+   (encrypted secrets), which are deployed by specific charts and decrypted by the
+   Sealed Secrets operator. But first, those Secrets must be seeded generated and
+   encrypted.
+
+   The LDAP server CA certificate goes into SealedSecret.
+
+   1. Load the `zeromq` container image required by Sealed Secret Generators.
+
+      > **NOTE:** A properly configured Docker or Podman environment is required.
+
+      ```bash
+      ncn-m001# /mnt/pitdata/${CSM_RELEASE}/hack/load-container-image.sh dtr.dev.cray.com/zeromq/zeromq:v4.0.5
+      ```
+
+   2. Re-encrypt the existing secrets:
+
+      ```bash
+      ncn-m001# /mnt/pitdata/prep/site-init/utils/secrets-reencrypt.sh customizations.yaml \
+      /mnt/pitdata/prep/site-init/certs/sealed_secrets.key /mnt/pitdata/prep/site-init/certs/sealed_secrets.crt
+      ```
+      
+4. Encrypt the static values in the customizations.yaml file after making changes.
+
+   The following command must be run within the site-init directory.
+
+   ```bash
+   ncn-m001# ./utils/secrets-seed-customizations.sh customizations.yaml
+   ```
+
+   Expected output looks similar to:
+
+      ```bash
+      Creating Sealed Secret keycloak-certs
+      Generating type static_b64...
+      Creating Sealed Secret keycloak-master-admin-auth
+      Generating type static...
+      Generating type static...
+      Generating type randstr...
+      Generating type static...
+      Creating Sealed Secret cray_reds_credentials
+      Generating type static...
+      Generating type static...
+      Creating Sealed Secret cray_meds_credentials
+      Generating type static...
+      Creating Sealed Secret cray_hms_rts_credentials
+      Generating type static...
+      Generating type static...
+      Creating Sealed Secret vcs-user-credentials
+      Generating type randstr...
+      Generating type static...
+      Creating Sealed Secret generated-platform-ca-1
+      Generating type platform_ca...
+      Creating Sealed Secret pals-config
+      Generating type zmq_curve...
+      Generating type zmq_curve...
+      Creating Sealed Secret munge-secret
+      Generating type randstr...
+      Creating Sealed Secret slurmdb-secret
+      Generating type static...
+      Generating type static...
+      Generating type randstr...
+      Generating type randstr...
+      Creating Sealed Secret keycloak-users-localize
+      Generating type static...
+      ```
+
+5. Resubmit the Kubernetes Job that runs the Keycloak localization tool.
+
+    1.  Re-apply the cray-keycloak-users-localize Helm chart with the updated customizations.yaml file.
+   
         ```bash
-        ncn-w001# utils/secrets-seed-customizations.sh customizations.yaml
+        ncn-m001# kubectl get job -n services -l app.kubernetes.io/name=cray-keycloak-users-localize \
```

Comment:
While the description is correct (that the cray-keycloak-users-localize Helm chart needs to be re-applied), the instructions here don't re-apply a Helm chart, they just re-submit the Job. Re-submitting the Job isn't going to pick up the updated values from customizations.yaml.

The Helm chart needs to be re-applied. I don't know how that's done but I assume it's possible.